src/PrestaShopBundle/DependencyInjection/Compiler/CommandAndQueryCollectorPass.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Collects all Commands & Queries and puts them into container for later processing.
+ */
+class CommandAndQueryCollectorPass implements CompilerPassInterface
```

Comment:
Doing it like this means that if we create a Command that is not being handled, it will not be displayed 🤔 ...

Well, if it cannot be handled, maybe we dont care 😅

Reply:
Maybe its kind of different responsibility that should be handled by some kind of dead-code linter?

Reply:
I dont know yet, the dead-code linter might be a good idea 😄something that would run in DEV environment and tell you "Eyh ! This Command is not handled, so if you or a module use it, this is going to fail, you should fix that"